File: client/src/app/models/order.ts
export interface Order {
    id: number;
    userId: string;
    shippingAddress: ShippingAddress;
    orderDate: string;
    orderItems: OrderItem[];
    subtotal: number;
    deliveryFee: number;
    orderStatus: string;
    total: number;
  }
  
  export interface OrderItem {
    productId: number;
    name: string;
    pictureUrl: string;
    price: number;
    quantity: number;
  }
  
  export interface ShippingAddress {
    fullname: string;
    address1: string;
    address2: string;
    city: string;
    state: string;
    zip: string;
    country: string;
  }

File: API/Entities/OrderAggregate/Order.cs
using System.ComponentModel.DataAnnotations;
using API.Entities.OrderAggregate;

namespace API.Entities
{
    public class Order
    {
        public int Id { get; set; }
        public string UserId { get; set; }

        [Required]
        public ShippingAddress ShippingAddress { get; set; }
        public DateTime OrderDate { get; set; } = DateTime.UtcNow;
        public List<OrderItem> OrderItems { get; set; }
        public decimal Subtotal { get; set; }
        public decimal DeliveryFee { get; set; }
        public OrderStatus OrderStatus { get; set; } = OrderStatus.Pending;
        public string PaymentIntentId { get; set; }
        public bool isDelivered { get; set; }

        public decimal GetTotal()
        {
            return Subtotal + DeliveryFee;
        }
    }
}